src/librustc_platform_intrinsics/lib.rs

```diff
@@ -37,60 +36,41 @@ pub enum IntrinsicDef {
     Named(&'static str),
 }
 
+
+static VOID: Type = Type::Void;
+
 static I8: Type = Type::Integer(true, 8, 8);
 static I16: Type = Type::Integer(true, 16, 16);
 static I32: Type = Type::Integer(true, 32, 32);
 static I64: Type = Type::Integer(true, 64, 64);
-static U8: Type = Type::Integer(false, 8, 8);
-static U16: Type = Type::Integer(false, 16, 16);
-static U32: Type = Type::Integer(false, 32, 32);
-static U64: Type = Type::Integer(false, 64, 64);
```

Comment:
From the error message it seems to be stating there is no 16byte width intrinsic defined with a type signature of `i16`. While there is `i8x16`, `i32x4`, and `i64x2`, these fail the type check. I'm digging into it today.